execution/client/client.go

```diff
@@ -64,17 +64,15 @@ func New(
 	telemetrySink TelemetrySink,
 	eth1ChainID *big.Int,
 ) *EngineClient {
+	ethClient := ethclientrpc.NewClient(
+		cfg.RPCDialURL.String(),
+		jwtSecret,
+		cfg.RPCJWTRefreshInterval,
+	)
 	return &EngineClient{
-		cfg:    cfg,
-		logger: logger,
-		Client: ethclient.New(
-			ethclientrpc.NewClient(
-				cfg.RPCDialURL.String(),
-				ethclientrpc.WithJWTSecret(jwtSecret),
-				ethclientrpc.WithJWTRefreshInterval(
-					cfg.RPCJWTRefreshInterval,
-				),
```

Comment:
just moved the instantiation of `ethClient` up and replace the options with actual parameter in its `New` function